src/proto/streams/counts.rs

```diff
@@ -46,6 +46,10 @@ impl Counts {
         self.peer
     }
 
+    pub fn has_streams(&self) -> bool {
+        self.num_send_streams != 0 || self.num_recv_streams != 0 || self.num_reset_streams != 0
```

Comment:
This probably doesn't need to check `num_reset_streams`, as that's just a count of "stream metadata we are holding on to temporarily to allow us to ignore incoming frames on a stream we just reset". The count is to just allow us to cap how many of those extra resources we keep in memory, but shouldn't affect if the connection is done, I believe.